osf/management/commands/handle_duplicate_files.py

```diff
@@ -0,0 +1,262 @@
+
+def inspect_duplicates(file_summary):
+    """
+    Inspects duplicates of a particular filetype and determines how many need to be resolved manually.
+    Outputs an array of files that can be successfully deleted.
+    """
+    to_remove = []
+    error_counter = 0
+
+    for duplicate_record in file_summary:
+        safe_to_remove = True
+        target_id, name, parent_id, file_type, path, count = duplicate_record
+        target = AbstractNode.objects.get(id=target_id)
+
+        files = return_basefilenode_values(
+            BaseFileNode.objects.filter(
+                name=name,
+                type=file_type,
+                target_object_id=target_id,
+                parent_id=parent_id,
+                _path=path
+            ).order_by('created')
+        )
+
+        preserved_file = files.first()
+
+        for next_file in files.exclude(_id=preserved_file['_id']):
+            safe_to_remove = resources_match(preserved_file, next_file)
+            if safe_to_remove:
+                to_remove.append(stage_removal(preserved_file, next_file))
+
+        if not safe_to_remove:
+            error_counter += 1
+            logger.info('Contact admins to resolve: target: {}, name: {}'.format(target._id, name))
+
+    logger.info('{}/{} errors to manually resolve.'.format(error_counter, len(file_summary)))
+    logger.info('Files that can be deleted without issue:')
+    for entry in to_remove:
+        logger.info('{}'.format(entry))
+    return to_remove
+
+def remove_duplicates(duplicate_array, file_type):
+    """
+    :param duplicate_array, expecting an array of dictionaries of format
+        [{'guid_to_repoint': guid, 'to_remove': file__id, 'preserving': file__id}]
+    """
+    for duplicate in duplicate_array:
+        guid = Guid.load(duplicate['guid_to_repoint'])
+        preserving = BaseFileNode.objects.get(_id=duplicate['preserving'])
+        to_remove = BaseFileNode.objects.get(_id=duplicate['to_remove'])
+
+        if guid:
+            guid.referent = preserving
+            guid.save()
+
+        to_remove.delete()
+    logger.info('Duplicates removed of type {}'.format(file_type))
+
+
+class Command(BaseCommand):
+    help = """
+    Searches for literal file or folder duplicates caused by dropped partial unique filename constraint
+
+    For osfstoragefiles - same name, same path, same parent,
+    same target, same type - versions all have identical SHA's
+
+    other addons -  have the same _history
+
+    Repoints guids, and then deletes all but one of the dupes.
+    """
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--dry_run',
+            type=bool,
+            default=False,
+            help='Run queries but do not repoint guids or delete dupes',
+        )
+        parser.add_argument(
+            '--file_type',
+            type=str,
+            required=True,
+            help='File type - must be in valid_file_types',
```

Comment:
Opinions on allowing `'all'` or `'all-files'`, for ease of use? I found this diff helpful
```
diff --git a/osf/management/commands/handle_duplicate_files.py b/osf/management/commands/handle_duplicate_files.py
index c759a0b232..0dd3f67ad0 100644
--- a/osf/management/commands/handle_duplicate_files.py
+++ b/osf/management/commands/handle_duplicate_files.py
@@ -232,7 +232,7 @@ class Command(BaseCommand):
             '--file_type',
             type=str,
             required=True,
-            help='File type - must be in valid_file_types',
+            help='File type - must be in valid_file_types or "all-files" or "all"',
         )

     # Management command handler
@@ -242,20 +242,31 @@ class Command(BaseCommand):
         dry_run = options['dry_run']
         file_type = options['file_type']

-        if file_type not in valid_file_types:
+        if file_type not in valid_file_types and file_type not in ['all', 'all-files'] :
             raise Exception('This is not a valid filetype.')

+        if file_type == 'all':
+            file_types = valid_file_types
+        elif file_type == 'all-files':
+            file_types = [t for t in valid_file_types if t != FOLDER]
+        else:
+            file_types = [file_type]
+
         if dry_run:
             logger.info('Dry Run. Data will not be modified.')

-        with connection.cursor() as cursor:
-            logger.info('Examining duplicates for {}'.format(file_type))
-            cursor.execute(FETCH_DUPLICATES_BY_FILETYPE, [file_type])
-            duplicate_files = cursor.fetchall()
-
-        to_remove = inspect_duplicates(duplicate_files)
-        if not dry_run:
-            remove_duplicates(to_remove, file_type)
+        for file_type in file_types:
+            type_start_time = datetime.datetime.now()
+            with connection.cursor() as cursor:
+                logger.info('Examining duplicates for {}'.format(file_type))
+                cursor.execute(FETCH_DUPLICATES_BY_FILETYPE, [file_type])
+                duplicate_files = cursor.fetchall()
+
+            to_remove = inspect_duplicates(duplicate_files)
+            if not dry_run:
+                remove_duplicates(to_remove, file_type)
+            type_finish_time = datetime.datetime.now()
+            logger.info('{} run time {}'.format(file_type, type_finish_time - type_start_time))

         script_finish_time = datetime.datetime.now()
         logger.info('Script finished time: {}'.format(script_finish_time))
```

Reply:
yes, great, just do most if it one go!